src/cmd-oscontainer

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+# An "oscontainer" is an ostree (archive) repository stuck inside
+# a Docker/OCI container at /srv/repo.  For more information,
+# see https://github.com/openshift/pivot
+#
+# This command manipulates those images.
+
+import gi
+gi.require_version('OSTree', '1.0')
+from gi.repository import GLib, Gio, OSTree
+import os,sys,json,shutil,argparse,subprocess,re,collections
+import tempfile,hashlib,gzip
+
+# See also flatpak_arch_to_oci_arch
+ostree_arch_to_oci = {
+    "x86_64": "amd64",
+    "aarch64": "arm64",
+    "i386": "386",
+}
+
+def run_get_json(args):
+    return json.loads(subprocess.check_output(args))
+
+# Given a container reference, pull the latest version, then extract the ostree
+# repo a new directory dest/repo.
+def oscontainer_extract(containers_storage, src, dest,
+                        tls_verify=True):
+    dest=os.path.realpath(dest)
+    subprocess.check_call(["ostree", "--repo="+dest, "refs"])
+    rootarg = '--root='+containers_storage
+    if not tls_verify:
+        tls_arg = '--tls-verify=false'
+    else:
+        tls_arg = '--tls-verify'
+    subprocess.check_call(['podman', rootarg, 'pull', tls_arg, src])
+    iid = run_get_json(['podman', rootarg, 'inspect', src])[0]['Id']
+    # /enoent since there's no binaries in this container, we're not
+    # actually going to run it, but `podman create` thinks we want to.
+    # The main thing this does is "materialize" the merged rootfs,
+    # so we can mount it.
+    cid = subprocess.check_output(['podman', rootarg, 'create', '--entrypoint=/enoent', iid]).strip()
```

Comment:
I guess this comment belongs somewhere around here instead: https://github.com/openshift/pivot/pull/19#discussion_r219303437.